gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java

```diff
@@ -395,21 +380,29 @@ private HttpClientRequest prepareRequest(final HttpServerRequest req, final Stri
             cRes.exceptionHandler(exception -> {
                 unpump.run();
                 error("Problem with backend: " + exception.getMessage(), req, targetUri);
-                req.response().setStatusCode(StatusCode.INTERNAL_SERVER_ERROR.getStatusCode());
-                req.response().setStatusMessage(StatusCode.INTERNAL_SERVER_ERROR.getStatusMessage());
-                try {
-                    ResponseStatusCodeLogUtil.info(req, StatusCode.INTERNAL_SERVER_ERROR, Forwarder.class);
-                    req.response().end(req.response().getStatusMessage());
-                } catch (IllegalStateException e) {
-                    // ignore because maybe already closed
-                }
+                respondError(req, StatusCode.INTERNAL_SERVER_ERROR);
             });
             req.connection().closeHandler((aVoid) -> {
                 unpump.run();
             });
         });
     }
 
+    private void respondError(HttpServerRequest req, StatusCode statusCode) {
+        try {
+            ResponseStatusCodeLogUtil.info(req, statusCode, Forwarder.class);
+
+            String msg = statusCode.getStatusMessage();
+            req.response()
+                    .setStatusCode(statusCode.getStatusCode())
+                    .setStatusMessage(msg)
+                    .end(msg);
+        } catch (IllegalStateException ex) {
+            // (nearly) ignore because underlying connection maybe already closed
+            RequestLoggerFactory.getLogger(Forwarder.class, req).info("IllegalStateException while sending error response for {}", req.uri(), ex);
```

Comment:
Is it correct to use _RequestLogger_ here?

Reply:
Not 100% sure - I don't completly understand this Logging-Infrastructure

Reply:
As you figured out correctly, the RequestLoggerFactory adds the request-id (when available) to the log statement to keep the link between log entry and request. You can refactor the RequestLoggerFactory as well if you want :-)
I agree to use this "log construct" here.